src/arithmetic.jl

```diff
@@ -469,12 +523,53 @@ for T in (:Taylor1, :TaylorN)
         return nothing
     end
 
+    @eval @inline function mul_scalar!(c::$T{T}, scalar::NumberNotSeries, a::$T{T}, b::$T{T}, k::Int) where {T<:Number}
+        if $T == Taylor1
+            @inbounds c[k] = a[0] * b[k]
+        else
+            @inbounds mul_scalar!(c[k], scalar, a[0], b[k])
+        end
+        @inbounds for i = 1:k
+            if $T == Taylor1
+                c[k] += a[i] * b[k-i]
+            else
+                mul_scalar!(c[k], scalar, a[i], b[k-i])
+            end
+        end
```

Comment:
Same comment wrt order of `for` and `if`

Reply:
This can be rewritten as
```julia
if $T == Taylor1
    @inbounds c[k] = a[0] * b[k]
    @inbounds for i = 1:k
        c[k] += a[i] * b[k-i]
     end
else
    @inbounds mul_scalar!(c[k], scalar, a[0], b[k])
    @inbounds for i = 1:k
        mul_scalar!(c[k], scalar, a[i], b[k-i])
    end
end
```